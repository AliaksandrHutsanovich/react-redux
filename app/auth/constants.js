export const formItemLayout = {
  labelCol: {
    xs: { span: 6 },
    sm: { span: 6 },
    md: { span: 6 },
    lg: { span: 6 },
  },
  wrapperCol: {
    xs: { span: 8, offset: 2 },
    sm: { span: 8, offset: 2 },
    md: { span: 8, offset: 2 },
    lg: { span: 8, offset: 2 },
  },
};

export const tailFormItemLayout = {
  wrapperCol: {
    xs: {
      span: 24,
      offset: 0,
    },
    sm: {
      span: 16,
      offset: 8,
    },
  },
};

export const emailValidationRules = [
  {
    type: 'email',
    message: 'The input is not valid E-mail!',
  },
  {
    required: true,
    message: 'Please input your E-mail!',
  },
];

export const passwordValidationRules = [
  {
    required: true,
    message: 'Please input your password!',
  },
];

export const confirmPasswordValidationRules = [
  {
    required: true,
    message: 'Please confirm your password!',
  },
];

export const fieldTitles = {
  password: 'password',
  confirm: 'confirm',
};

export const formTypes = {
  signUp: 'signUp',
  logIn: 'logIn',
};

export const isValidateByFormType = {
  [formTypes.signUp]: true,
  [formTypes.logIn]: false,
};

export const buttonTitleByFormType = {
  [formTypes.signUp]: 'Register',
  [formTypes.logIn]: 'Log in',
};
