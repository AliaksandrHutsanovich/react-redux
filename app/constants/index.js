const OPERATION_TITLES = {
  ADD_NEW_SUBCATEGORY: 'Add new subcategory',
  EDIT_CATEGORY: 'Edit category',
  DELETE_CATEGORY: 'Delete category',
  ADD_NEW_CATEGORY: 'Add new category',
  ADD_NEW_TASK: 'Add new task',
};

export default OPERATION_TITLES;
