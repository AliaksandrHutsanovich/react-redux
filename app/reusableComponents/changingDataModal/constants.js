import OPERATION_TITLES from '../../constants';

const FOOTER_BY_OPERATION_TITLE = {
  [OPERATION_TITLES.EDIT_CATEGORY]: [],
  [OPERATION_TITLES.ADD_NEW_SUBCATEGORY]: [],
};

export default FOOTER_BY_OPERATION_TITLE;
