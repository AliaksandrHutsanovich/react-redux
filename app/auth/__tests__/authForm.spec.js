import React from 'react';
import { shallow } from 'enzyme';
import { Input, Form } from 'antd';
import AuthForm from '../authForm';
import { formTypes } from '../constants';

const { Password } = Input;


describe('<AuthForm />', () => {
  const signUp = shallow(<AuthForm type={formTypes.signUp} />);
  const preventDefault = jest.fn();

  it('Sign up form should render properly', () => {
    expect(signUp).toMatchSnapshot();
  });

  it('Log in form should render properly', () => {
    const logIn = shallow(<AuthForm type={formTypes.logIn} />).shallow();
    expect(logIn).toMatchSnapshot();
  });

  it('form should work', () => {
    signUp.shallow().find(Input).at(0).simulate('change', {
      target: {
        value: 'gu-84@ret.tu',
      },
    });
    signUp.shallow().find(Password).at(0).simulate('change', {
      target: {
        value: '12345',
      },
    });
    signUp.shallow().find(Password).at(1).simulate('change', {
      target: {
        value: '12345',
      },
    });

    signUp.shallow().find(Password).at(1).simulate('blur', {
      target: {
        value: '12345',
      },
    });

    signUp.shallow().find(Form).prop('onSubmit')({ preventDefault });

    expect(preventDefault).toHaveBeenCalled();
    expect(signUp.prop('form').getFieldsValue(['email', 'password'])).toEqual({
      email: 'gu-84@ret.tu',
      password: '12345',
    });
  });
});
