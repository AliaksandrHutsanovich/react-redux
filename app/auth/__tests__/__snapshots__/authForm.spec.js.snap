// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthForm /> Log in form should render properly 1`] = `
<Form
  autoComplete="off"
  colon={true}
  hideRequiredMark={false}
  labelCol={
    Object {
      "lg": Object {
        "span": 6,
      },
      "md": Object {
        "span": 6,
      },
      "sm": Object {
        "span": 6,
      },
      "xs": Object {
        "span": 6,
      },
    }
  }
  layout="horizontal"
  onSubmit={[Function]}
  wrapperCol={
    Object {
      "lg": Object {
        "offset": 2,
        "span": 8,
      },
      "md": Object {
        "offset": 2,
        "span": 8,
      },
      "sm": Object {
        "offset": 2,
        "span": 8,
      },
      "xs": Object {
        "offset": 2,
        "span": 8,
      },
    }
  }
>
  <FormItem
    hasFeedback={false}
    key="0"
    label="E-mail"
  >
    <Input
      data-__field={
        Object {
          "name": "email",
        }
      }
      data-__meta={
        Object {
          "name": "email",
          "originalProps": Object {
            "type": "text",
          },
          "ref": null,
          "rules": Array [
            Object {
              "message": "The input is not valid E-mail!",
              "type": "email",
            },
            Object {
              "message": "Please input your E-mail!",
              "required": true,
            },
          ],
          "trigger": "onChange",
          "validate": Array [
            Object {
              "rules": Array [
                Object {
                  "message": "The input is not valid E-mail!",
                  "type": "email",
                },
                Object {
                  "message": "Please input your E-mail!",
                  "required": true,
                },
              ],
              "trigger": Array [
                "onChange",
              ],
            },
          ],
          "valuePropName": "value",
        }
      }
      id="register_email"
      onChange={[Function]}
      type="text"
    />
  </FormItem>
  <FormItem
    hasFeedback={true}
    key="1"
    label="Password"
  >
    <Password
      action="click"
      data-__field={
        Object {
          "name": "password",
        }
      }
      data-__meta={
        Object {
          "name": "password",
          "originalProps": Object {
            "action": "click",
            "inputPrefixCls": "ant-input",
            "prefixCls": "ant-input-password",
            "visibilityToggle": true,
          },
          "ref": null,
          "rules": Array [
            Object {
              "message": "Please input your password!",
              "required": true,
            },
          ],
          "trigger": "onChange",
          "validate": Array [
            Object {
              "rules": Array [
                Object {
                  "message": "Please input your password!",
                  "required": true,
                },
              ],
              "trigger": Array [
                "onChange",
              ],
            },
          ],
          "valuePropName": "value",
        }
      }
      id="register_password"
      inputPrefixCls="ant-input"
      onChange={[Function]}
      prefixCls="ant-input-password"
      visibilityToggle={true}
    />
  </FormItem>
  <FormItem
    hasFeedback={false}
    wrapperCol={
      Object {
        "sm": Object {
          "offset": 8,
          "span": 16,
        },
        "xs": Object {
          "offset": 0,
          "span": 24,
        },
      }
    }
  >
    <Button
      block={false}
      ghost={false}
      htmlType="submit"
      loading={false}
      type="primary"
    >
      Log in
    </Button>
  </FormItem>
</Form>
`;

exports[`<AuthForm /> Sign up form should render properly 1`] = `
<AuthForm
  form={
    Object {
      "getFieldDecorator": [Function],
      "getFieldError": [Function],
      "getFieldInstance": [Function],
      "getFieldProps": [Function],
      "getFieldValue": [Function],
      "getFieldsError": [Function],
      "getFieldsValue": [Function],
      "isFieldTouched": [Function],
      "isFieldValidating": [Function],
      "isFieldsTouched": [Function],
      "isFieldsValidating": [Function],
      "isSubmitting": [Function],
      "resetFields": [Function],
      "setFields": [Function],
      "setFieldsInitialValue": [Function],
      "setFieldsValue": [Function],
      "submit": [Function],
      "validateFields": [Function],
      "validateFieldsAndScroll": [Function],
    }
  }
  type="signUp"
/>
`;
